tools/tpm2_policycountertimer.c

```diff
@@ -0,0 +1,198 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+
+#include <stdlib.h>
+
+#include "files.h"
+#include "tpm2.h"
+#include "tpm2_nv_util.h"
+#include "tpm2_options.h"
+#include "tpm2_policy.h"
+
+typedef struct tpm_policycountertimer_ctx tpm_policycountertimer_ctx;
+struct tpm_policycountertimer_ctx {
+    //Inputs
+    const char *session_path;
+    tpm2_session *session;
+
+    const TPM2B_OPERAND operand_b;
+
+    UINT16 offset;
+
+    TPM2_EO operation;
+
+    //Output
+    const char *policy_digest_path;
+};
+
+static tpm_policycountertimer_ctx ctx = {
+    .operation = TPM2_EO_EQ,
+};
+
+static bool on_arg(int argc, char **argv) {
+
+    if (argc > 1) {
+        LOG_ERR("Specify one argument - The Comparison-Operation");
+        return false;
+    }
+
+        if (!strcmp(argv[0], "eq")) {
+            ctx.operation = TPM2_EO_EQ;
+        } else if (!strcmp(argv[0], "neq")) {
+            ctx.operation = TPM2_EO_NEQ;
+        } else if (!strcmp(argv[0], "sgt")) {
+            ctx.operation = TPM2_EO_SIGNED_GT;
+        } else if (!strcmp(argv[0], "ugt")) {
+            ctx.operation = TPM2_EO_UNSIGNED_GT;
+        } else if (!strcmp(argv[0], "slt")) {
+            ctx.operation = TPM2_EO_SIGNED_LT;
+        } else if (!strcmp(argv[0], "ult")) {
+            ctx.operation = TPM2_EO_UNSIGNED_LT;
+        } else if (!strcmp(argv[0], "sge")) {
+            ctx.operation = TPM2_EO_SIGNED_GE;
+        } else if (!strcmp(argv[0], "uge")) {
+            ctx.operation = TPM2_EO_UNSIGNED_GE;
+        } else if (!strcmp(argv[0], "sle")) {
+            ctx.operation = TPM2_EO_SIGNED_LE;
+        } else if (!strcmp(argv[0], "ule")) {
+            ctx.operation = TPM2_EO_UNSIGNED_LE;
+        } else if (!strcmp(argv[0], "bs")) {
+            ctx.operation = TPM2_EO_BITSET;
+        } else if (!strcmp(argv[0], "bc")) {
+            ctx.operation = TPM2_EO_BITCLEAR;
+        } else {
+            LOG_ERR("Invalid Comparison-Operation specified.");
+            return false;
+        }
+
+    return true;
+}
+
+static bool process_input_operand_b(char *value) {
+
+    char *input_file = strcmp("-", value) ? value : NULL;
+    bool result;
+
+    if (input_file) {
+        result = files_get_file_size_path(value,
+                (long unsigned *) &ctx.operand_b.size);
+    }
+    if (input_file && !result) {
+        return false;
+    }
+
+    result = files_load_bytes_from_buffer_or_file_or_stdin(NULL, input_file,
+            (UINT16 *)&ctx.operand_b.size,
+            (unsigned char *)&ctx.operand_b.buffer);
```

Comment:
get rid of that ampersand on the buffer thats why its complaining. You shouldn't need to cast these.